packages/echo-core/src/services/analytics/analyticsModule.ts

```diff
@@ -0,0 +1,149 @@
+import { BaseError, ForbiddenError, ValidationError } from '@equinor/echo-base';
+import { IExceptionTelemetry, SeverityLevel } from '@microsoft/applicationinsights-web';
+import { EchoEnv } from '../../EchoEnv';
+import { appWithModuleName, eventNameToString } from './appInSightsLogic';
+import { AnalyticsEvent, EventName, propertyTypes } from './appInsightTypes';
+import { appInsightsInstance } from './appInsightWrapper';
+import OfflineTracker from './offlineTracker';
+
+// Based on Client Analytics Strategy
+// https://github.com/equinor/Echo/pull/14/files#diff-1e78f3b1fd220c5050d7497a73068a5e
+// Examples:
+// echo.AssetModel.Unloaded
+// echo.AssetModel.Loaded
+// ep.Document.Loaded
+// ep_xld.Documented.Opened <-- Module xld in app/service ep opened a document
+// - Object
+//   - The object that has an action performed. Such as `AssetModel` and `MeasuringTool`, they should be generic. We do **not** want both `echo.AssetModelJohanSverdrup.Loaded` and `echo.AssetModelGudrun.Loaded`. Separating which context the event was in will be a part of the **event data**.
+// - Action
+//   - The action performed on the event. This is written in past tense. (Viewed, Disabled, Enabled, Started, etc)
+
+let instCode = '';
+let userCompany = '';
+
+/**
+ * Sets the instCode logged with all events, set globally for analytics modules
+ * @param plantInstCode instCode logged with all events
+ */
+export function analyticsSetInstCode(plantInstCode: string): void {
+    instCode = plantInstCode;
+}
+
+/**
+ * Sets the company logged with all events, set globally for analytics modules
+ * @param company company logged with all events
+ */
+export function analyticsSetUserCompany(company: string): void {
+    userCompany = company.startsWith('X-') ? company.replace('X-', '') : 'Equinor';
+}
+
+/**
+ * Analytics module used to log data to application Insight
+ */
+export class AnalyticsModule {
+    moduleName: string;
+    offlineTracker: OfflineTracker;
+    constructor(moduleName: string) {
+        this.moduleName = moduleName;
+
+        const offlineThresholdSeconds = 20;
+        this.offlineTracker = new OfflineTracker(offlineThresholdSeconds, !navigator.onLine);
+        window.addEventListener('online', () => this.trackOnline);
+        window.addEventListener('offline', () => this.trackOffline);
+    }
+
+    private trackOnline(): void {
+        const evt = this.offlineTracker.setOnline();
```

Comment:
Maybe it's just me: I think it would help readability if this would be actually called `event`.

Reply:
good point, we should not use abbreviations